src/main/java/nextstep/optional/User.java

```diff
@@ -33,7 +35,14 @@ public static boolean ageIsInRange1(User user) {
     }
 
     public static boolean ageIsInRange2(User user) {
-        return false;
+        return Optional.ofNullable(user).map(elm -> checkAge(elm.age))
+                .orElse(false);
+    }
+
+    private static boolean checkAge(Integer userAge) {
+        return Optional.ofNullable(userAge)
+                .map(age -> (userAge >= 30 && userAge <= 45) ? true : false)
+                .orElse(false);
```

Comment:
map은 요소를 변환하는 역할을 담당하고 있기에 비교 조건은 filter를 이용하면 좋을 것 같아요!
```suggestion
                .filter(age -> age >= 30 && age <= 45)
                .isPresent();
```

Reply:
넵 다음 스텝에 반영해두겠습니다 👍